pkg/query-service/rules/thresholdRule.go

```diff
@@ -1062,56 +703,91 @@ func (r *ThresholdRule) GetSelectedQuery() string {
 	return ""
 }
 
-// query looks if alert condition is being
-// satisfied and returns the signals
 func (r *ThresholdRule) buildAndRunQuery(ctx context.Context, ts time.Time, ch clickhouse.Conn) (Vector, error) {
 	if r.ruleCondition == nil || r.ruleCondition.CompositeQuery == nil {
 		r.SetHealth(HealthBad)
+		r.SetLastError(fmt.Errorf("no rule condition"))
 		return nil, fmt.Errorf("invalid rule condition")
 	}
 
-	// var to hold target query to be executed
-	var queries map[string]string
-	var err error
-
-	// fetch the target query based on query type
-	if r.ruleCondition.QueryType() == v3.QueryTypeBuilder {
-
-		queries, err = r.prepareBuilderQueries(ts, ch)
+	params := r.prepareQueryRange(ts)
+	err := r.populateTemporality(ctx, params, ch)
+	if err != nil {
+		r.SetHealth(HealthBad)
+		zap.L().Error("failed to set temporality", zap.String("rule", r.Name()), zap.Error(err))
+		return nil, fmt.Errorf("internal error while setting temporality")
+	}
 
-		if err != nil {
-			zap.L().Error("failed to prepare metric queries", zap.String("ruleid", r.ID()), zap.Error(err))
-			return nil, fmt.Errorf("failed to prepare metric queries")
+	if params.CompositeQuery.QueryType == v3.QueryTypeBuilder {
+		// check if any enrichment is required for logs if yes then enrich them
+		if logsv3.EnrichmentRequired(params) {
+			// Note: Sending empty fields key because enrichment is only needed for json
+			// TODO: Add support for attribute enrichment later
```

Comment:
please create an issue for this if not already created.

reason:- 

* sometimes alert is created but the key is indexed later, but not enriching it uses the non indexed key